testsuite/module-publish/src/packages/simple/sources/token_v1.move

```diff
@@ -0,0 +1,184 @@
+        let minted_tokens = &mut borrow_global_mut<MinterConfig>(creator_address).minted_tokens;
+        table::add(minted_tokens, user_address, option::some(token));
+    }
+
+    /// Make the tokendata name unique by appending a ` #` and the index
+    fun build_token_name(token_prefix: String, index: u64): String {
+        string::append_utf8(&mut token_prefix, b" #");
+        string::append(&mut token_prefix, to_string<u64>(&index));
+        token_prefix
+    }
+
+    fun token_v1_create_token_data(
+        creator: &signer,
+        token_uri: String,
+        tokendata_name: String,
+        maximum: u64
+    ): TokenDataId {
+        let collection_name = string::utf8(COLLECTION_NAME);
+        let tokendata_description = string::utf8(TOKEN_DESCRIPTION);
+        let creator_address = signer::address_of(creator);
+
+        // tokan max mutable: true
+        // token URI mutable: true
+        // token description mutable: true
+        // token royalty mutable: true
+        // token properties mutable: true
+        let token_mutate_config = token::create_token_mutability_config(&vector<bool>[ true, true, true, true, true ]);
+
+        let property_keys = vector[];
+        let property_values = vector[];
+        let property_types = vector[];
+
+        token::create_tokendata(
+            creator,
+            collection_name,
+            tokendata_name,
+            tokendata_description,
+            maximum,
+            token_uri,
+            creator_address,
+            ROYALTY_POINTS_DENOMINATOR,
+            ROYALTY_POINTS_NUMERATOR,
+            token_mutate_config,
+            property_keys,
+            property_values,
+            property_types
+        )
+    }
+
+
+    fun mint_nft_sequential(creator_address: address) : (signer, TokenId) acquires MinterConfig {
+        let resource_signer = get_signer(creator_address);
+        let resource_signer_address = signer::address_of(&resource_signer);
+
+        // Make the tokendata name unique by appending a ` #` and the current supply + 1
+        let current_supply_opt = token::get_collection_supply(resource_signer_address, string::utf8(COLLECTION_NAME));
+        let index = option::extract(&mut current_supply_opt) + 1;
+        let tokendata_name = build_token_name(string::utf8(TOKEN_NAME), index);
+
+        let tokendata_id = token_v1_create_token_data(&resource_signer, string::utf8(TOKEN_URI), tokendata_name, 1);
+        let token_id = token::mint_token(&resource_signer, tokendata_id, 1);
+        (resource_signer, token_id)
+    }
+
+    fun mint_nft_parallel(user: &signer, creator_address: address) : (signer, TokenId) acquires MinterConfig {
+        let resource_signer = get_signer(creator_address);
+        let token_name = build_token_name(to_string<address>(&signer::address_of(user)), account::get_sequence_number(signer::address_of(user)));
```

Comment:
it's not visible from the diff view, but this is the change in the contract, to make sure we can mint multiple NFTs to the same destination account